src/attr/_make.py

```diff
@@ -709,12 +731,45 @@ def _add_method_dunders(self, method):
         return method
 
 
+_CMP_DEPRECATION = (
+    "The usage of `cmp` is deprecated and will be removed on or after "
+    "2021-06-01.  Please use `eq` and `order` instead."
+)
+
+
+def _determine_eq_order(cmp, eq, order):
+    """
+    Validate the combination of *cmp*, *eq*, and *order*. Derive the effective
+    values of eq and order.
+    """
+    if cmp is not None and any((eq is not None, order is not None)):
+        raise ValueError("Don't mix `cmp` with `eq' and `order`.")
+
+    # cmp takes precedence due to bw-compatibility.
+    if cmp is not None:
+        warnings.warn(_CMP_DEPRECATION, DeprecationWarning, stacklevel=3)
+
+        return cmp, cmp
+
+    # Both on by default.
```

Comment:
Is this comment right? It seems like if eq is False and order is None you still get order = False?

I guess "both on by default" means "both on unless you touch either one", which yeah ok nevermind.

Reply:
I have clarified the comment.